docs/api-opentelemetry.asciidoc

```diff
@@ -84,6 +84,37 @@ you can use the OpenTelemetry API to get ahold of the current span and call `set
 Span.current().setAttribute("foo", "bar");
 ----
 
+[float]
+[[otel-set-behavioral-attribute]]
+==== Customize span tracing
+
+We utilize the `setAttribute()` API not only to <<otel-set-attribute, add custom metadata>>, but also as a way to customize some
+special tracing features through corresponding custom attributes listed below. Such attributes are not added to span metadata. For example:
+
+[source,java]
+----
+Span.current().setAttribute("elastic.otel.nondiscardable", true);
+----
+
+The `co.elastic.apm.agent.opentelemetry.sdk.BehavioralAttributes` utility class provides supported attributes as keys. Using them is
```

Comment:
We'd have to publish a dedicated module for users to be able to use this class, right?

Reply:
Yes, this makes no sense. I thought of making it nicer as an `enum` and realized that a big incentive for users to use OTel is avoid vendor-locking, so not very appealing.
This also worth considering about the usage of such custom "behavioral" attributes that we do not add as span metadata, but other OTel implementations would

Reply:
Ideally yes, but maybe we could just simply support both:
- with a dependency on the not-yet-published class
- with an attribute that just follows the same naming convention (which allows to not have a strong dependency)

So, I suggest that for short term we only support the naming convention, then we handle this properly with a published artifact when deprecating our API and/or when we have defined more attributes in sync with other agents.

From what I remember in OTel the semantic attributes are defined in yaml, then code is generated for every platform, while we might not need to go that far and keep doing it manually, just choosing a proper name for the artifact might require some alignment with other agents.

Reply:
The thing is that the "discardable" logic is something that's only in the Java agent. Other agents don't implement `span_min_duration`, only `exit_span_min_duration` and therefore don't need the more complex discardable handling.

Reply:
I'd remove the section that mentions `co.elastic.apm.agent.opentelemetry.sdk.BehavioralAttributes` for now as there's no way for users to get a dependency on a module that contains this class.

Reply:
Sure, which means it makes more sense to be reverted back to be constant strings.
We'll followup later if we feel there is a need then with exposing somehow